### PR TITLE
feat(backend): adds utility function for clearing flags on a model

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a utility function for clearing flags on an arbitrary model. This is used during migrations if we have an old flag that's no longer in use. Tests have been added.

### CODE REVIEW

File: src/sentry/utils/migrations.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+def clear_flag(Model, flag_name, flag_attr_name="flags"):
+    """
+    This function is used to clear an existing flag value for all items in a given model
+    """
+    for item in RangeQuerySetWrapperWithProgressBar(Model.objects.all()):
+        flags = getattr(item, flag_attr_name)
+        if flags[flag_name]:
+            # clear the flag
+            new_flag_value = flags & ~(getattr(Model, flag_attr_name)[flag_name])
+            setattr(item, flag_attr_name, new_flag_value)
```
Comment (wedamija) [inline src/sentry/utils/migrations.py:13]: Can we do `setattr(item, flag_attr_name, False)` here?

File: src/sentry/utils/migrations.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def clear_flag(Model, flag_name, flag_attr_name="flags"):
```
Comment (wedamija) [inline src/sentry/utils/migrations.py:4]: It might be worth making `flag_name` a list, in case we want to reset mulitple at the same time

**Review by wedamija — COMMENTED** [review]
Overall looks good to me

File: src/sentry/utils/migrations.py (hunk 1 above)
Comment (scefali) [reply src/sentry/utils/migrations.py:13]: @wedamija wouldn't that just do something like
```
Model.flags = False
```

Comment (wedamija) [reply src/sentry/utils/migrations.py:13]: Right, I think I mean
`getattr(item, flag_attr_name)[flag_name] = False`

File: src/sentry/utils/migrations.py (hunk 2 above)
Comment (scefali) [reply src/sentry/utils/migrations.py:4]: @wedamija hmmm...I could do that but then the logic becomes a bit more complex. I'm not convinced it's worth the tradeoff (I don't expect flag clearing to happen _that_ often).

File: src/sentry/utils/migrations.py (hunk 1 above)
Comment (scefali) [reply src/sentry/utils/migrations.py:13]: If so, that def won't work lol

Comment (scefali) [reply src/sentry/utils/migrations.py]: @wedamija switching to `.filter().update()` so I'm using a different way of doing it

**Review by wedamija — APPROVED** [review]

**Pull request merged**